oak-query-spi/src/main/java/org/apache/jackrabbit/oak/spi/query/QueryIndex.java

```diff
@@ -361,17 +364,32 @@ default boolean logWarningForPathFilterMismatch() {
          * This method can be used for communicating any messages which should be logged if this plan is selected for execution.
          * The messages are returned as a map whose key indicates log level and value is a list of messages against that
          * log level.
+
+         * @deprecated use {@link #getAdditionalLogMessages()}  instead
          * @return map containing log messages.
          */
+        @Deprecated(forRemoval = true)
         default Map<Level, List<String>> getAdditionalMessages() {
-            return Collections.emptyMap();
+            LoggerFactory.getLogger(QueryIndex.IndexPlan.class).warn("use of deprecated API - this method is going to be removed in future Oak releases - see OAK-10768 for details");
```

Comment:
Shouldn't we add the name of the new API which is replacing it

Reply:
Certainly we could, but this is covered by the JavaDoc, isn't it?